scripts/install_deps.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+GLFW_URL="https://github.com/glfw/glfw/archive/refs/tags/3.4.tar.gz"
+GLEW_URL="https://github.com/nigels-com/glew/releases/download/glew-2.2.0/glew-2.2.0.tgz"
+ASSIMP_URL="https://github.com/assimp/assimp/archive/refs/tags/v5.4.0.tar.gz"
+
+
+LIB_PREFIX=~/libs
+ARCHIVE="sources"
+
+clear () {
+	rm -rf ~/.tmp_deps
+	mkdir ~/.tmp_deps
+	cd ~/.tmp_deps
+}
+
+cmake_install () {
+	cmake .
+	cmake --build .
+	cmake --install . --prefix "$LIB_PREFIX"
+}
+
+download_lib () {
+	wget $1 -O "$ARCHIVE"
+	tar xf "$ARCHIVE"
+	rm -rf "$ARCHIVE"	
+}
+
+clear
+download_lib "$GLFW_URL"
+cd *
+cmake_install
+
+clear
+download_lib "$GLEW_URL"
+cd */build/cmake
+cmake_install
+
+clear
+download_lib "$ASSIMP_URL"
+cd *
+cmake_install
+
+clear
```

Comment:
```suggestion
#!/bin/bash

GLFW_URL="https://github.com/glfw/glfw/archive/refs/tags/3.4.tar.gz"
GLEW_URL="https://github.com/nigels-com/glew/releases/download/glew-2.2.0/glew-2.2.0.tgz"
ASSIMP_URL="https://github.com/assimp/assimp/archive/refs/tags/v5.4.0.tar.gz"

LIB_PREFIX=~/libs
ARCHIVE="sources"
TMP_DIR=~/.tmp_deps

clear () {
	rm -rf $TMP_DIR
	mkdir $TMP_DIR
	cd $TMP_DIR
}

cmake_install () {
	cmake .
	cmake --build .
	cmake --install . --prefix "$LIB_PREFIX"
}

download_lib () {
	wget $1 -O "$ARCHIVE"
	tar xf "$ARCHIVE"
	rm -rf "$ARCHIVE"	
}

urls=("$GLFW_URL" "$GLEW_URL" "$ASSIMP_URL")

for url in "${urls[@]}"; do
	clear
	download_lib "$url"

	# For GLEW we need to call cmake_install from a different directory
	if [[ $url == "$GLEW_URL" ]]; then
		cd */build/cmake
	else
		cd *
	fi

	cmake_install
	cd -
	
done

clear
```

Reply:
Soooo i dont know about that soultion with if and loop, right now we have 3 libs here which we build in a similar way, but I doubt that we can have universal steps for building all the libraries, we will need for this project in the future thats why I decided to just leave the building steps "hardcoded" for each library. But idk, what do you think?

Reply:
In this case we can extract the code, which is in the for loop, to the function with two parameters - url and install path - at least we save some lines.